.github/workflows/py-tests.yml

```diff
@@ -35,7 +35,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        py-version: ['3.7', '3.8', '3.9', '3.10']
+        py-version: [3.8', '3.9', '3.10']
```

Comment:
since I we are updating actions, let's remove 3.7 support already. Official deprecation is for 1.2 so we can start applying 3.8+ lang features